CMakeLists.txt

```diff
@@ -0,0 +1,58 @@
+# Copyright 2016-2021 Lawrence Livermore National Security, LLC and other
+# IREP Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: MIT
+
+#To use this cmakefile, lua needs to be in your PATH.
+
+cmake_minimum_required(VERSION 3.1)
+project(irep)
+
+# Find lua and figure out bin dir
+find_package(Lua REQUIRED)
+string(REGEX REPLACE "/include$" "/bin" LUA_BIN ${LUA_INCLUDE_DIR})
+
+# Set up IREP_GENERATE varaible
+include(${CMAKE_CURRENT_SOURCE_DIR}/share/irep/irep-config.cmake)
+
+# Be sure to include lua headers
+include_directories(
+    ${CMAKE_CURRENT_SOURCE_DIR}
+    ${LUA_INCLUDE_DIR}
+)
+
+# ir_std.f & ir_extern.f is generated from coresponding
+# headers using IREP_GENERATE
+add_custom_command(
+    OUTPUT ir_std.f
+    COMMAND
+        ${CMAKE_COMMAND} -E env PATH="${LUA_BIN}:$ENV{PATH}"
+        ${IREP_GENERATE} --mode fortran ${CMAKE_CURRENT_SOURCE_DIR}/ir_std.h > ir_std.f
+)
+
+add_custom_command(
+    DEPENDS ir_std.f
+    OUTPUT ir_extern.f
+    COMMAND
+        ${CMAKE_COMMAND} -E env PATH="${LUA_BIN}:$ENV{PATH}"
+        ${IREP_GENERATE} --mode fortran ${CMAKE_CURRENT_SOURCE_DIR}/ir_extern.h > ir_extern.f
+)
+
+# libIR.a is mostly basic functions from irep.c.
+# ir_extern.f, & ir_std.f are very much like wkt headers.
+# This ensures their tables are in libIR.a. Note that they are automatically included
+# in index libraries by irep-config.cmake
+add_library(
+    IR STATIC
+    irep.c
+    ${CMAKE_CURRENT_BINARY_DIR}/ir_extern.f
+    ${CMAKE_CURRENT_BINARY_DIR}/ir_std.f
+)
+
+# Copy libIR.a to current source directory
+add_custom_command(
+    TARGET IR
+    POST_BUILD
+    COMMAND
+        ${CMAKE_COMMAND} -E copy libIR.a ${CMAKE_CURRENT_SOURCE_DIR}/libIR.a
```

Comment:
This doesn't actually install to the right location (it installs to the source directory, not `CMAKE_INSTALL_PREFIX`). 

Note that there are 3 locations we care about:
* `CMAKE_CURRENT_SOURCE_DIR`: where the source is (this should be considered read-only.
* `CMAKE_CURRENT_BINARY_DIR`: where the build is happening (i.e. where generated code gets spit out)
* `CMAKE_INSTALL_PREFIX`: where we want everything to go in the end

The `install()` command will handle the details of all this.  Look at how it's done in the CMakeLists files in the `exo` build for examples.